services/vpm/vpm-version.service.js

```diff
@@ -0,0 +1,79 @@
+import Joi from 'joi'
+import { optionalUrl } from '../validators.js'
+import { latest, renderVersionBadge } from '../version.js'
+import { BaseJsonService, NotFound } from '../index.js'
+
+const queryParamSchema = Joi.object({
+  repository_url: optionalUrl.required(),
+  include_prereleases: Joi.equal(''),
+}).required()
+
+const schema = Joi.object({
+  packages: Joi.object()
+    .pattern(
+      /./,
+      Joi.object({
+        versions: Joi.object().pattern(/./, Joi.object()).min(1).required(),
+      }).required()
+    )
+    .required(),
+}).required()
+
+export default class VpmVersion extends BaseJsonService {
+  static category = 'version'
+
+  static route = {
+    base: 'vpm/v',
+    pattern: ':packageId',
+    queryParamSchema,
+  }
+
+  static examples = [
+    {
+      title: 'VPM Package Version',
+      namedParams: {
+        packageId: 'com.vrchat.udonsharp',
+      },
+      queryParams: {
+        repository_url: 'https://packages.vrchat.com/curated?download',
+      },
+      staticPreview: renderVersionBadge({ version: '1.1.6' }),
+    },
+    {
+      title: 'VPM Package Version (including prereleases)',
+      namedParams: {
+        packageId: 'com.vrchat.udonsharp',
+      },
+      queryParams: {
+        repository_url: 'https://packages.vrchat.com/curated?download',
+        include_prereleases: null,
+      },
+      staticPreview: renderVersionBadge({ version: '1.1.6' }),
```

Comment:
should I use fake prerelease version like `1.1.7-beta.1` for static preview?

Reply:
I think it is fine as it stands